### PR TITLE
#167190405 Follow and Unfollow users

### PR DESCRIPTION
**Pull request by chiboycalix**
#### What are the relevant pivotal tracker stories?
[#167190405](https://www.pivotaltracker.com/story/show/167190405)

### CODE REVIEW

File: src/actions/profileAction.js [hunk 1]
```diff
@@ -46,11 +46,14 @@ export const getArticlesFailure = (error) => ({
   error,
 });
 
-export const getProfile = () => async (dispatch) => {
+export const getProfile = (username = null) => async (dispatch) => {
```
Comment (obayomi96) [inline src/actions/profileAction.js]: Hi @chiboycalix , is there a reason why the username is set to the value of null from the get profile function parameter?

File: src/actions/profileAction.js [hunk 2]
```diff
@@ -63,19 +66,25 @@ export const editProfile = (payload) => async (dispatch) => {
...
+      timer: 3000,
     });
     return dispatch(editProfileSuccess(response.data.profile));
   } catch (error) {
     return dispatch(editProfileFailure(error.response.data));
   }
 };
 
-export const getArticles = () => async (dispatch) => {
+export const getArticles = (authorId = null) => async (dispatch) => {
```
Comment (obayomi96) [inline src/actions/profileAction.js]: And here

File: src/views/Articles/ReadArticle/ReadArticle.js [hunk 3]
```diff
@@ -70,6 +128,31 @@ export class ReadArticle extends Component {
...
+    this.setState(prevState => ({
+      ...prevState,
+      isFollowing: false,
+    }));
+  };
+
+  handleProfile = () => {
+    const { article: { author: { username } } } = this.props;
+    // eslint-disable-next-line react/destructuring-assignment
+    return this.props.history.push(`/profiles/${username}`);
```
Comment (Lundii) [inline src/views/Articles/ReadArticle/ReadArticle.js]: Nice work Nonso. I think 
```
const { history } = this.props;
```
will solve the react/destructuring-assignment eslint flag

File: src/views/ProfilePage/Profile.js [hunk 4]
```diff
@@ -1,9 +1,12 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable no-nested-ternary */
 /* eslint-disable react/require-default-props */
 /* eslint-disable jsx-a11y/no-static-element-interactions */
```
Comment (Lundii) [inline src/views/ProfilePage/Profile.js:2]: Also, I think the `react/destructuring-assignment` and the `require-default-props` warning can easily be solved.

Comment (chiboycalix) [reply src/views/ProfilePage/Profile.js:2]: thank you for this. I'll fix that

File: src/actions/profileAction.js (hunk 2 above)
Comment (chiboycalix) [reply src/actions/profileAction.js]: It has been removed

**Review by Lundii — APPROVED** [review]
LGTM

**Review by encodedBicoding — APPROVED** [review]
LGTM.Nice work Nonso

**Review by Hadeneekeh — APPROVED** [review]
LGTM

**Review by obayomi96 — APPROVED** [review]
LGTM

**Review by chuxmykel — APPROVED** [review]
Looks good to me.

**Pull request merged**